File: bonsai_interfaces/src/main/java/de/unibi/citec/clf/bonsai/actuators/ObjectDetectionActuator.java
package de.unibi.citec.clf.bonsai.actuators;

import de.unibi.citec.clf.bonsai.core.object.Actuator;
import de.unibi.citec.clf.btl.List;
import de.unibi.citec.clf.btl.data.object.ObjectLocationData;
import java.util.concurrent.ExecutionException;

/**
 *
 * @author saschroeder
 */
public interface ObjectDetectionActuator extends Actuator {
    List<ObjectLocationData> detect() throws InterruptedException, ExecutionException;
}

File: bonsai_interfaces/src/main/java/de/unibi/citec/clf/bonsai/actuators/ObjectDetectionActuator.kt
package de.unibi.citec.clf.bonsai.actuators

import de.unibi.citec.clf.bonsai.core.`object`.Actuator
import de.unibi.citec.clf.btl.List
import de.unibi.citec.clf.btl.data.`object`.ObjectShapeData
import de.unibi.citec.clf.btl.data.geometry.BoundingBox3D
import java.io.IOException

import java.util.concurrent.ExecutionException
import java.util.concurrent.Future

/**
 *
 * @author lruegeme
 */
interface ObjectDetectionActuator : Actuator {
    @Throws(IOException::class)
    fun detectObjects(roi: BoundingBox3D? = null): Future<List<ObjectShapeData>>

    @Throws(IOException::class)
    fun detectSurface() : Future<BoundingBox3D>
}
